python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -1843,6 +1843,58 @@ void basic_imported_symbols() {
     assertThat(((UnionType) mathModule.resolveMember("acos").get()).candidates()).allMatch(FunctionType.class::isInstance);
   }
 
+  @Test
+  void isInstanceTests() {
+    var xType = lastExpression("""
+      def foo():
+        x = 10
+        if isinstance(x, Foo):
+          ...
+        x
+      """).typeV2();
+
+    Assertions.assertThat(xType).isSameAs(PythonType.UNKNOWN);
+
+    xType = lastExpression("""
+      def foo():
+        x = 10
+        if isinstance(x):
+          ...
+        x
+      """).typeV2();
+
+    Assertions.assertThat(xType.unwrappedType()).isSameAs(INT_TYPE);
+
+    xType = lastExpression("""
+      def foo():
+        x = 10
+        if isinstance(x.b, Foo):
+          ...
+        x
+      """).typeV2();
+
+    Assertions.assertThat(xType.unwrappedType()).isSameAs(INT_TYPE);
+
+    xType = lastExpression("""
+      def foo():
```

Comment:
In the test you took this from, `x: int` is a parameter of `foo`. Without it, I don't think the test makes sense since `x` is going to be unknown no matter what.